SingularityService/src/main/java/com/hubspot/singularity/data/LoggingCuratorFramework.java

```diff
@@ -0,0 +1,312 @@
+package com.hubspot.singularity.data;
+
+import com.google.inject.Inject;
+import java.util.Timer;
+import java.util.TimerTask;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+import java.util.concurrent.TimeUnit;
+import org.apache.curator.CuratorZookeeperClient;
+import org.apache.curator.framework.CuratorFramework;
+import org.apache.curator.framework.WatcherRemoveCuratorFramework;
+import org.apache.curator.framework.api.CreateBuilder;
+import org.apache.curator.framework.api.CuratorListener;
+import org.apache.curator.framework.api.DeleteBuilder;
+import org.apache.curator.framework.api.ExistsBuilder;
+import org.apache.curator.framework.api.GetACLBuilder;
+import org.apache.curator.framework.api.GetChildrenBuilder;
+import org.apache.curator.framework.api.GetConfigBuilder;
+import org.apache.curator.framework.api.GetDataBuilder;
+import org.apache.curator.framework.api.ReconfigBuilder;
+import org.apache.curator.framework.api.RemoveWatchesBuilder;
+import org.apache.curator.framework.api.SetACLBuilder;
+import org.apache.curator.framework.api.SetDataBuilder;
+import org.apache.curator.framework.api.SyncBuilder;
+import org.apache.curator.framework.api.UnhandledErrorListener;
+import org.apache.curator.framework.api.transaction.CuratorMultiTransaction;
+import org.apache.curator.framework.api.transaction.CuratorTransaction;
+import org.apache.curator.framework.api.transaction.TransactionOp;
+import org.apache.curator.framework.imps.CuratorFrameworkState;
+import org.apache.curator.framework.listen.Listenable;
+import org.apache.curator.framework.schema.SchemaSet;
+import org.apache.curator.framework.state.ConnectionStateErrorPolicy;
+import org.apache.curator.framework.state.ConnectionStateListener;
+import org.apache.curator.utils.EnsurePath;
+import org.apache.zookeeper.Watcher;
+import org.apache.zookeeper.server.quorum.flexible.QuorumVerifier;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class LoggingCuratorFramework implements CuratorFramework {
+  private final CuratorFramework curator;
+
+  private final ConcurrentMap<String, Integer> counters;
+
+  private final Long interval = 60000L * 3; // 3 minutes
+
+  private static final Logger LOG = LoggerFactory.getLogger(
+    LoggingCuratorFramework.class
+  );
+
+  private class ClearCounter extends TimerTask {
+
+    @Override
+    public void run() {
+      logAndClear();
+    }
+  }
+
+  @Inject
+  public LoggingCuratorFramework(CuratorFramework curator) {
+    this.curator = curator;
+    counters = new ConcurrentHashMap<>();
+    Timer timer = new Timer();
+
+    timer.schedule(new ClearCounter(), interval, interval);
```

Comment:
We should use a ScheduledExecutorService instead of a Timer here (better thread control, exception handling).

There's a good example in the SingularityHealthchecker https://github.com/HubSpot/Singularity/blob/master/SingularityService/src/main/java/com/hubspot/singularity/scheduler/SingularityHealthchecker.java#L93
https://github.com/HubSpot/Singularity/blob/master/SingularityService/src/main/java/com/hubspot/singularity/scheduler/SingularityHealthchecker.java#L277